File: Contests/2019-2020/Bronze/Dec/gymnastics.cpp
#include <bits/stdc++.h>
#pragma GCC optimize(3)
using namespace std;
typedef long long ll;
const int MAXN = 1005;

int n, k, a[15][25], t, cnt;
int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    freopen("gymnastics.in", "r", stdin);
    freopen("gymnastics.out", "w", stdout);
    cin >> k >> n;
    for (int i = 1; i <= k; ++i) {
        for (int j = 1; j <= n; ++j) {
            cin >> t;
            a[i][t] = j;
        }
    }
    for (int i = 1; i <= n; ++i) {
        for (int j = i+1; j <= n; ++j) {
            int c = 0;
            for (int l = 1; l <= k; ++l) {
                if (a[l][i] > a[l][j]) ++c;
            }
            if (c==0 || c==k) ++cnt;
        }
    }
    cout << cnt << endl;
    return 0;
}

File: Contests/2019-2020/Bronze/Dec/gymnastics.py
import sys
sys.stdin = open('gymnastics.in', 'r')
sys.stdout = open('gymnastics.out', 'w')

k, n = map(int, input().split())
a = []
for i in range(k):
    a.append([int(i) for i in input().split()])

res = 0
for i in range(1, n+1):
    for j in range(i+1, n+1):
        cnt = 0
        for l in range(k):
            if a[l].index(i) < a[l].index(j):
                cnt += 1
        if cnt == k or cnt == 0: 
            res += 1
print(res)